tmitchell/exercise.py

```diff
@@ -0,0 +1,281 @@
+                count = 1
+        if count == 0:
+            print("Could not find any entries matching that description")
+
+    def man_add_person(self):
+        """Request input from user, create new entry based on that then add to phonebook."""
+        print("Enter new entry's name:")
+        new_name = input("")
+        if self.check_duplicate(new_name) == 0:
+            print("An entry with that name already exists,")
+            print("please try again with a different name")
+        else:
+            print("Enter new entry's home number:")
+            new_hnumber = input("")
+            print("Enter new entry's mobile number:")
+            new_mnumber = input("")
+            new_person = Person(new_name, new_hnumber, new_mnumber)
+            self.people.append(new_person)
+
+    def delete_person(self):
+        """Request input from user, identify then delete appropriate entry."""
+        print("Enter exact name of entry to be deleted")
+        delrow = input("")
+        index = self.identify_by_name(delrow)
+        if index == "null":
+            print("Delete failed! Returning to main menu")
+        else:
+            print("Deleting %s" % (self.people[index].name))
+            del self.people[index]
+
+    def edit_person(self):
+        """Request input from user, identify entry and then edit it based on further input."""
+        print("Enter exact name of entry to be edited")
+        edited = input("")
+        index = self.identify_by_name(edited)
+        if index == "null":
+            print("Edit failed! Returning to main menu")
+        else:
+            print("Current entry Name: %s" % (self.people[index].name))
+            print("Enter new name: (leave blank to remain the same)")
+            new_name = input("")
+            dupe = self.check_duplicate(new_name)
+            if (dupe == 0) & (new_name != self.people[index].name):
+                print("An entry with that name already exists,")
+                print("please try again with a different name")
+            else:
+                self.people[index].edit(new_name)
+
+    def identify_by_name(self, name):
+        """Identify entry then return entry's position in book."""
+        count = 0
+        for i in self.people:
+            if name == i.name:
+                return count
+            count += 1
+        print("No entry found with that name")
+        return "null"
+
+    def check_duplicate(self, name):
+        """Check for entrys with the provided name, returns 0 if already exists or 1 otherwise."""
+        for i in self.people:
+            if name == i.name:
+                return 0
+        return 1
+
+    def order_entries(self):
+        """Sort phonebook alphabetically(ish)."""
+        sortlist = []
+        for i in self.people:
+            if not sortlist:
+                sortlist.append(i)
+            else:
+                counter = len(sortlist)-1
+                while counter >= 0:
```

Comment:
Classic CS - this is a double loop: can you think of a more efficient way to do this?